tests/test_intron.py

```diff
@@ -0,0 +1,140 @@
+"""Test intron."""
+from ngs_test_utils import testcase
+
+from intron import (
+    intronCounter_v2_stranded,
+    make_conserved_gtf,
+    make_conserved_intron_gtf,
+)
+
+
+class IntronTestCase(testcase.NgsTestCase):
+    """Test intron."""
+
+    def setUp(self):
+        self.outdir = self.get_tmp_dir()
+        self.pkw = {"is_paired": True, "is_proper_pair": True}
+
+    def test_intron_ucsc(self):
+        gtf = self.make_gtf(
+            [
+                dict(
+                    seqname="chr1",
+                    feature="exon",
+                    start=99,
+                    end=199,
+                    strand="+",
+                    transcript_id="NM_1",
+                    gene_id="G1",
+                ),
+                dict(
+                    seqname="chr1",
+                    feature="exon",
+                    start=299,
+                    end=399,
+                    strand="+",
+                    transcript_id="NM_1",
+                    gene_id="G1",
+                ),
+            ]
+        )
+        bam = self.make_bam(
+            chroms=[("chr1", 1000)],
+            segments=[
+                dict(
+                    qname="r1",
+                    pos=150,
+                    pnext=350,
+                    cigar=[(0, 75)],
+                    is_read1=True,
+                    is_reverse=True,
+                    **self.pkw,
+                ),
+                dict(
+                    qname="r1",
+                    pos=350,
+                    pnext=150,
+                    cigar=[(0, 75)],
+                    is_read2=True,
+                    mate_is_reverse=True,
+                    **self.pkw,
+                ),
+            ],
+        )
+
+        ann1 = self.get_filename(extension="gtf")
+        ann2 = self.get_filename(extension="gtf")
+        result = self.get_filename(extension="gtf")
+
+        make_conserved_gtf.process_gtf(gtf, ann1, source="UCSC")
```

Comment:
The tests are insufficient. I cannot comment the relevant lines of code directly because they haven't been modified in this PR. Briefly, too long genes are not removed because we add locus position after popping the gene and thus reinstate it's index. Please fix the bug and write a test.

Crosschecks show that long genes may not be a UCSC artifact. They map well onto the corresponding ENSEMBL genes, e.g. [DAB1](http://www.ensembl.org/Homo_sapiens/Gene/Summary?g=ENSG00000173406;r=1:56994778-58546734) and [AGBL4](http://www.ensembl.org/Homo_sapiens/Gene/Summary?g=ENSG00000186094;r=1:48532854-50023954). Are we still sure that we want to remove them?

We don't do a check if a gene with the same ID appears on more than one chromosome and more than one strand. Instead of relying on gene length check to remove non-unique `gene_ids` I suggest that we implement a multiindex `(gene_id, chromosome, strand)` in construction of dict `d` and try to work with non-unique `gene_ids`.